modules/apps/segments/segments-web/package.json

```diff
@@ -0,0 +1,65 @@
+{
+	"csfConfig": {
+		"js": {
+			"lint": {
+				"env": {
+					"browser": true,
+					"es6": true,
+					"node": true
+				},
+				"parser": "babel-eslint",
+				"parserOptions": {
+					"ecmaVersion": 7
+				}
+			}
+		}
+	},
+	"dependencies": {
+		"prop-types": "^15.6.2",
+		"react": "^16.6.3",
+		"react-dom": "^16.6.3"
+	},
+	"description": "",
+	"devDependencies": {
+		"@babel/cli": "^7.0.0",
+		"@babel/core": "^7.1.6",
+		"@babel/plugin-proposal-class-properties": "^7.1.0",
+		"@babel/preset-env": "^7.1.6",
+		"@babel/preset-react": "^7.0.0",
+		"babel-core": "^7.0.0-bridge.0",
+		"babel-eslint": "^10.0.1",
+		"babel-loader": "^8.0.4",
+		"babel-preset-env": "^1.7.0",
+		"babel-preset-react": "6.24.1",
+		"check-source-formatting": "^2.0.13",
+		"eslint": "^5.8.0",
+		"eslint-config-liferay": "^2.0.18",
+		"eslint-plugin-babel": "^5.3.0",
+		"eslint-plugin-react": "^7.11.1",
+		"eslint-plugin-sort-destructure-keys": "^1.1.0",
+		"jest": "^23.6.0",
+		"liferay-npm-bundler": "^2.0.0",
+		"liferay-npm-bundler-preset-liferay-dev": "^1.1.0",
+		"odata-v4-parser": "^0.1.29",
+		"webpack": "^4.27.0",
+		"webpack-cli": "^3.1.2"
+	},
+	"jest": {
+		"setupTestFrameworkScriptFile": "<rootDir>/test/setup.js",
+		"testMatch": [
+			"<rootDir>/test/js/**/*.js"
+		]
+	},
+	"main": "js/index.es.js",
+	"name": "segments-web",
+	"private": true,
+	"scripts": {
+		"build": "babel --source-maps --verbose -d classes/META-INF/resources src/main/resources/META-INF/resources && liferay-npm-bundler",
+		"build:libs": "webpack",
```

Comment:
Continue the conversation from the other PR.

Is the idea that whenever you want an updated version of `odata-v4-parser` you would re-run this script?

Also, for using `odata-v4-parser` when you import it normally like `import * from 'odata-v4-parser'` does that pick it up from the lib folder or from node_modules?

Reply:
It's mainly there to show how that file was compiled.

When using `odata-v4-parser` I import the compiled file directly, so it's added as a global `window.oDataParser`. https://github.com/bryceosterhaus/liferay-portal/blob/89d5a606121cd65872247993d67cf1a3f272388f/modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/ClayODataQueryBuilder.es.js#L19

Reply:
gotcha! thanks for the explanation